lib/services/local/notifications.go

```diff
@@ -0,0 +1,291 @@
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	userNotificationStateService, err := generic.NewServiceWrapper[*notificationsv1.UserNotificationState](backend, types.KindUserNotificationState, notificationsStatePrefix, services.MarshalUserNotificationState, services.UnmarshalUserNotificationState)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	userLastSeenNotificationService, err := generic.NewServiceWrapper[*notificationsv1.UserLastSeenNotification](backend, types.KindUserLastSeenNotification, notificationsUserLastSeenPrefix, services.MarshalUserLastSeenNotification, services.UnmarshalUserLastSeenNotification)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &NotificationsService{
+		log:                             logrus.WithFields(logrus.Fields{trace.Component: "notifications:local-service"}),
+		userNotificationService:         userNotificationService,
+		globalNotificationService:       globalNotificationService,
+		userNotificationStateService:    userNotificationStateService,
+		userLastSeenNotificationService: userLastSeenNotificationService,
+	}, nil
+}
+
+// ListDatabases returns a paginated list of notifications which match a user, including both user-specific and global ones.
+func (s *NotificationsService) ListNotificationsForUser(ctx context.Context) ([]*notificationsv1.Notification, string, error) {
+	// TODO: rudream - implement listing notifications for a user with filtering/matching
+	return []*notificationsv1.Notification{}, "", nil
+}
+
+// CreateUserNotification creates a user-specific notification.
+func (s *NotificationsService) CreateUserNotification(ctx context.Context, username string, notification *notificationsv1.Notification) (*notificationsv1.Notification, error) {
+	err := services.ValidateNotification(notification)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	notification.Kind = types.KindNotification
+	notification.Version = types.V1
+
+	if notification.Spec == nil {
+		notification.Spec = &notificationsv1.NotificationSpec{}
+	}
+
+	if notification.Spec.Id == "" {
+		// Generate uuidv7 ID.
+		uuid, err := uuid.NewV7()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		notification.Spec.Id = uuid.String()
+	}
+	// We set this to the UUID as well because the service adapter uses `getName()` to determine the backend key to use when storing the notification.
+	notification.Metadata.Name = notification.Spec.Id
+
+	// Set default expiry to 30 days from now.
+	if notification.Metadata.Expires == nil {
+		now := time.Now()
+		futureTime := now.Add(30 * 24 * time.Hour)
+		timestamp := timestamppb.New(futureTime)
+
+		notification.Metadata.Expires = timestamp
+	}
+
+	if notification.Spec.Created == nil {
+		notification.Spec.Created = timestamppb.New(time.Now())
+	}
```

Comment:
> What point in time is created supposed to capture? The time that a user created the notification or the time that Teleport stored the object in the backend?

Both, because creating a notification _is_ storing it in the backend.

I think it actually does make sense for us to always set it here and validate that it isn't provided by the consumer, since there is no scenario where `Created` should be represent anything other than the time the notification was created (ie. stored in the backend).